Set Debian's default locale to be C.UTF-8

TheRealFalcon opened the pull request:
## Proposed Commit Message
<!-- Include a proposed commit message because all PRs are squash merged -->

```
Set Debian's default locale to be C.UTF-8

When no locale is found in /etc/default/locale, cloud-init was
defaulting to generating en_US.UTF-8. This can be an expensive operation
and isn't necessary. Instead, change the default to C.UTF-8 which
generally shouldn't require generation.

Also add some logging to make it more obvious where cloud-init is
getting the locale information from.

LP: #2038945
```

## Additional Context
<!-- If relevant -->

## Test Steps
<!-- Please include any steps necessary to verify (and reproduce if
this is a bug fix) this change on a live deployed system,
including any necessary configuration files, user-data,
setup, and teardown. Scripts used may be attached directly to this PR. -->

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes
that apply. -->
 - [ ] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/contributing.html)
 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

mwhudson commented on cloudinit/distros/debian.py:
```diff
@@ -54,7 +54,7 @@ def __init__(self, name, cfg, paths):
         # should only happen say once per instance...)
         self.osfamily = "debian"
-        self.default_locale = "en_US.UTF-8"
+        self.default_locale = "c.UTF-8"
```
I'm fairly sure it doesn't matter but can you spell this "C.UTF-8" please!

blackboxsw commented on cloudinit/distros/debian.py:
```diff
@@ -66,10 +66,21 @@ def get_locale(self):
+            )
+            return self.system_locale
+        else:
+            LOG.debug(
```
Changeset looks good, but we have no unittest coverage for this path of the conditional

blackboxsw replied:
tested on images w/out /etc/default/locale. Success regenerating locales in absence of system defaults

TheRealFalcon replied:
Looking at this more, I may move the logging around. In the case where no locale has been specified, `get_locale` is going to get called twice so we'll see duplicate logs

blackboxsw reviewed (approved):
Changset looks good to set locale defaults. Let's also get review from Noah on this as debian could prefer images to retain a default of `en_US.UTF-8`  if the image itself didn't provide a default in /etc/default/locale (or by proxy /etc/locale.conf).

TheRealFalcon commented:
@nmeyerhans Would you mind reviewing this?

TheRealFalcon commented:
@blackboxsw I moved the log messages so that they don't log multiple times. I also added unit tests. One commit is converting the tests to pytest. The other commit is the actual test changes.

blackboxsw reviewed (approved):
Approved w/ unittest coverage and logging messages are only written once

nmeyerhans reviewed (approved):
👍 to changing the default locale; it is consistent with what we set in the Debian cloud images, so should be a no-op in most cases.

holmanb commented:
> Also, tests require a fixed version of #4499 . Without it we get no logs from caplog.

+1 Merged
